rooms.js

```diff
@@ -639,29 +639,22 @@ class GlobalRoom extends BasicRoom {
 		let roomsData = {official: [], pspl: [], chat: [], userCount: this.userCount, battleCount: this.battleCount};
 		for (const room of this.chatRooms) {
 			if (!room) continue;
+			if (room.parent) continue;
 			if (room.isPrivate && !(room.isPrivate === 'voice' && user.group !== ' ')) continue;
+			let roomData = {
+				title: room.title,
+				desc: room.desc,
+				userCount: room.userCount,
+			};
+			if (room.subRooms && room.subRooms.size) roomData.subRooms = room.getSubRooms().map(room => room.title);
```

Comment:
`room.subRooms.size` is redundant, right?

...you _are_ setting `room.subRooms = null` when you remove the last subroom, right?????